service/src/main/kotlin/xyz/lebkuchenfm/domain/history/HistoryRepository.kt

```diff
@@ -3,9 +3,9 @@ package xyz.lebkuchenfm.domain.history
 import com.github.michaelbull.result.Result
 
 interface HistoryRepository {
-    suspend fun insert(history: HistoryEntry): Result<HistoryEntry, HistoryRepositoryError>
+    suspend fun insert(history: HistoryEntry): Result<HistoryEntry, InsertHistoryEntryError>
 }
 
-sealed class HistoryRepositoryError {
-    data object UnknownError : HistoryRepositoryError()
+sealed class InsertHistoryEntryError {
```

Comment:
WDYT about not specifying it to `Insert` Error?
```kotlin
sealed class HistoryRepositoryError {
     data object WritingHistoryError : HistoryRepositoryError()
     // because soon we might want to
     data object ReadingHistoryError : HistoryRepositoryError()
 }
 ```

Reply:
No, each operation should have its own error type. If we don't do that we will make matching on error type useless (why do we need do handle writing error, when we're reading?).